internal/distro/fedora33/distro_test.go

```diff
@@ -0,0 +1,344 @@
+			inputSize:  10 * gigaByte,
+			outputSize: 10 * gigaByte,
+		},
+		{
+			name:       "vhd",
+			inputSize:  10*gigaByte - 1,
+			outputSize: 10 * gigaByte,
+		},
+	}
+
+	distro := fedora33.New()
+	arch, err := distro.GetArch("x86_64")
+	if assert.NoError(t, err) {
+		for _, mapping := range sizeMap {
+			imgType, err := arch.GetImageType(mapping.name)
+			if assert.NoError(t, err) {
+				size := imgType.Size(mapping.inputSize)
+				assert.Equalf(t, mapping.outputSize, size, "Image type: %s, input size: %d, expected: %d, got: %d",
+					mapping.name, mapping.inputSize, mapping.outputSize, size)
+			}
+		}
+	}
+}
+
+func TestImageType_BasePackages(t *testing.T) {
+	pkgMaps := []struct {
+		name               string
+		basePackages       []string
+		bootloaderPackages []string
+		excludedPackages   []string
+		bootable           bool
+	}{
+		{
+			name: "ami",
+			basePackages: []string{
+				"@Core",
+				"chrony",
+				"kernel",
+				"selinux-policy-targeted",
+				"langpacks-en",
+				"libxcrypt-compat",
+				"xfsprogs",
+				"cloud-init",
+				"checkpolicy",
+				"net-tools",
+			},
+			bootloaderPackages: []string{
+				"dracut-config-generic",
+				"grub2-pc",
+			},
+			excludedPackages: []string{
+				"dracut-config-rescue",
+				"geolite2-city",
+				"geolite2-country",
+				"zram-generator-defaults",
+			},
+			bootable: true,
+		},
+		{
+			name: "openstack",
+			basePackages: []string{
+				"@Core",
+				"chrony",
+				"kernel",
+				"selinux-policy-targeted",
+				"spice-vdagent",
+				"qemu-guest-agent",
+				"xen-libs",
+				"langpacks-en",
+				"cloud-init",
+				"libdrm",
+			},
+			bootloaderPackages: []string{
+				"dracut-config-generic",
+				"grub2-pc",
+			},
+			excludedPackages: []string{
+				"dracut-config-rescue",
+				"geolite2-city",
+				"geolite2-country",
+				"zram-generator-defaults",
+			},
+			bootable: true,
+		},
+	}
+	distro := fedora33.New()
+	arch, err := distro.GetArch("x86_64")
+	assert.NoError(t, err)
+
+	for _, pkgMap := range pkgMaps {
+		imgType, err := arch.GetImageType(pkgMap.name)
+		assert.NoError(t, err)
+		basePackages, excludedPackages := imgType.Packages(blueprint.Blueprint{})
+		assert.Equalf(
+			t,
+			append(pkgMap.basePackages, pkgMap.bootloaderPackages...),
+			basePackages,
+			"image type: %s",
+			pkgMap.name,
+		)
+		assert.Equalf(t, pkgMap.excludedPackages, excludedPackages, "image type: %s", pkgMap.name)
+	}
+}
+
+func TestDistro_Manifest(t *testing.T) {
+	distro_test_common.TestDistro_Manifest(t, "../../../test/cases/", "fedora_33*", fedora33.New())
+}
+
+func TestFedora33_ListArches(t *testing.T) {
+	distro := fedora33.New()
+	arches := distro.ListArches()
+	assert.Equal(t, []string{"aarch64", "x86_64"}, arches)
+}
+
+func TestFedora33_GetArch(t *testing.T) {
+	distro := fedora33.New()
+	arches := []struct {
+		name          string
+		errorExpected bool
+	}{
+		{
+			name: "x86_64",
+		},
+		{
+			name: "aarch64",
+		},
+		{
+			name:          "foo-arch",
```

Comment:
Some extra whitespace here.

Reply:
go fmt seems to want this so it aligns with the `errorExpected: true` field.

Reply:
oh i see, looked weird to my eyes. :+1: